programs/zstdcli.c

```diff
@@ -419,9 +471,20 @@ static int readSizeTFromCharChecked(const char** stringPtr, size_t* value)
  *  Will also modify `*stringPtr`, advancing it to position where it stopped reading.
  *  Note : function will exit() program if digit sequence overflows */
 static size_t readSizeTFromChar(const char** stringPtr) {
-    static const char errorMsg[] = "error: numeric value overflows size_t";
+
     size_t result;
-    if (readSizeTFromCharChecked(stringPtr, &result)) { errorOut(errorMsg); }
+    const char* startChar;
+    int returnVal;
+
+    startChar = *stringPtr;
+    
+    returnVal = readSizeTFromCharChecked(stringPtr, &result);
+    if (returnVal == 1) { 
+        errorOut("error: numeric value overflows size_t");
+    }
+    else if (returnVal == 2) {
+        DISPLAY("warning: poorly formed parameter of %s, resetting to default \n", startChar);
+    }
```

Comment:
If we encounter an error we want to `errorOut`.

Reply:
No worries, can do